tensornetwork/network_components.py

```diff
@@ -1325,6 +1324,132 @@ def flatten_all_edges(nodes: Iterable[BaseNode]) -> List[Edge]:
   return flattened_edges
 
 
+def _split_trace_edge(edge: Edge,
+                      shape: Tuple[int, ...],
+                      new_edge_names: Optional[List[Text]] = None,
+                      ) -> List[Edge]:
+  """Split trace edges into single edge.
+
+  Args:
+    edge: Trace edge to split.
+    shape: Tuple of integers used to split trace edge into multiple edges.
+    new_edge_names: Optional names of the new edges created.
+
+  Returns:
+    A list of new edges where the product of the dimensions of the new
+    edges corresponds to the dimension of the edge before splitting.
+  """
+  node = edge.node1  # We are in the trace case, so this is the only node.
+  backend = node.backend
+  # Permute until edge axes to be split are at the back and reshape.
+  perm_back = [min(edge.axis1, edge.axis2)]
+  perm_back += [max(edge.axis1, edge.axis2)]
+  perm_front = set(range(len(node.edges))) - set(perm_back)
+  perm_front = sorted(perm_front)
+  node.reorder_axes(perm_front + perm_back)
+  unaffected_shape = backend.shape_tuple(node.tensor)[:len(perm_front)]
```

Comment:
I would recommend doing `backend.shape` as tensors that have an undefined shape at runtime need to have their shapes as tenors.